crates/sui-core/src/gateway_types.rs

```diff
@@ -1228,7 +1228,7 @@ pub enum SuiEvent {
 }
 
 impl SuiEvent {
-    fn try_from(event: Event, resolver: &impl GetModule) -> Result<Self, anyhow::Error> {
+    pub fn try_from(event: Event, resolver: &impl GetModule) -> Result<Self, anyhow::Error> {
```

Comment:
I think we do not need the resolver: `type` and `fields` can be found in the `event` obj: https://github.com/MystenLabs/sui/blob/cdfd8dc6913a50584399649780d0273a0c4dcfe5/crates/sui-types/src/event.rs#L71-L73

Reply:
the reseller is needed to convert the `MoveObject` which contains BCS bytes, into `SuiMoveObject`, which contain parsed `SuiMoveStruct` for a better json representation.